### ISSUE TITLE
Unrelated errors in latitude detection (sometimes)

### ISSUE THREAD

**Issue by Yash-10** [poliastro/poliastro#1290]
Since the latitude detector uses the `cartesian_to_ellipsoidal` method directly and also because `cartesian_to_ellipsoidal` calculates the longitude and height at the same time, sometimes the latitude detector returns a zero division error in [this](https://github.com/poliastro/poliastro/blob/main/src/poliastro/core/spheroid_location.py#L185) line:

```py
h = z / np.sin(lat) - (1 - e2) * v
```

Since this is not an error due to latitude calculation but an error in a different calculation (here, `h` the height), it might be confusing and/or unnecessary to raise this error when using the latitude event.

**Reproducible example**
```py
tofs = [10, 1000, 3000, 4000] << u.s
orbit = Orbit.circular(Earth, 150 * u.km)

thresh_lat = 60 * u.deg
latitude_cross_event = LatitudeCrossEvent(orbit, thresh_lat, terminal=False)
events = [latitude_cross_event]

rr, _ = cowell(
    Earth.k,
    orbit.r,
    orbit.v,
    tofs,
    events=events,
    f=f,
)

```

**Possible ideas**
- Make three separate methods, each for `lat`, `lon`, and `h` to prevent unrelated error pop-up
- Also, if spheroid locations can be converted into a "class", it could be easy to access the lat, on, and h attributes.

**Comment by astrojuanlu** [comment]
If `np.sin(lat) == 0`, it means that `lat == 0`. But therefore, `z == 0`, and so `z / np.sin(lat)` should have some sultion? I vote for trying to handle this case in a numerically correct way

**Comment by Yash-10** [comment]
@astrojuanlu Do you think this issue was solved by #1292?

**Comment by astrojuanlu** [comment]
Indeed!
